Core/Model/Base/ModelClass.php

```diff
@@ -263,7 +263,35 @@ public function save()
      */
     public function test()
     {
-        return true;
+        $status = true;
+
+        foreach ($this->getModelFields() as $key => $value) {
+            $fieldType = $value['type'];
+            $fieldNullable = $value['is_nullable'];
+
+            if (empty($value['default']) && $fieldNullable === 'NO' && $this->{$key} === null && $this->{$key} !== $this->primaryColumnValue()) {
+                self::$miniLog->alert(self::$i18n->trans('field-can-not-be-null', ['%fieldName%' => $key]));
+                $status = false;
+            }
+
+            // Force to match type from supported types, because from API we receive always a string
```

Comment:
Please remove data type conversions. API problema must be solved in API. Model is not a garbage collector.